src/main/java/com/yahoo/bullet/parsing/expressions/LazyExpression.java

```diff
@@ -0,0 +1,133 @@
+package com.yahoo.bullet.parsing.expressions;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import com.yahoo.bullet.common.Configurable;
+import com.yahoo.bullet.common.Initializable;
+import com.yahoo.bullet.typesystem.Type;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.util.Arrays.asList;
+
+/**
+ * Use "Lazy" as a placeholder name because it rolls off the tongue.
+ *
+ * Lazy expressions are currently used in queries for filters and projections.
+ *
+ * A filter is simply a lazy expression. We accept a record iff it the expression evaluates to true (with an explicit cast to Boolean), e.g.
+ *
+ * "filter": { "left": {"field": "bcookie"}, "right": {"value": "123456", "type": "STRING"}, "op": "EQUALS" }
```

Comment:
Use another value besides bcookie

Reply:
Actually it's because it's open source that we don't want our internal fields out here

Reply:
That's what I meant .-.

Reply:
Oh I read your response as a statement not a question.